troubleshooting/ETW/Readme.md

```diff
@@ -127,43 +127,69 @@ This file will not exceed the configured max size and will be circularly overwri
 double click on the process to pop up Properties dialog, and find "Current directory" in "Image" tab.
 Internally, the SDK looks for the configuration file located in [GetCurrentDirectory](https://docs.microsoft.com/dotnet/api/system.io.directory.getcurrentdirectory),
 and then [AppContext.BaseDirectory](https://docs.microsoft.com/dotnet/api/system.appcontext.basedirectory).
 You can also find the exact directory by calling these methods from your code.
 
-**To disable self-diagnostics**, delete the configuration file.
+    Example:
 
-Example: 
-```json
-{
-    "LogDirectory": ".",
-    "FileSize": 1024,
-    "LogLevel": "Error"
-}
-```
+    ```json
+    {
+        "LogDirectory": ".",
+        "FileSize": 1024,
+        "LogLevel": "Error"
+    }
+    ```
+
+2. For the case that users are not allowed to create the `ApplicationInsightsDiagnostics.json` config file
+at the above mentioning locations due to hosting enviornment limiations, configuration can be set in via enviornment variables.
+
+    Example:
+
+    ```csharp
+    {
+        // "LogDirectory" must be set if users want application insights to read configuration from envrionment variables. 
+        Environment.SetEnvironmentVariable("LogDirectory", ".");
+
+        // "FileSize" is an optional parameter, i.e. if "LogDirectory" was set, the default value for FileSize is 1 MiB.
+        Environment.SetEnvironmentVariable("FileSize", "1024");
+
+        // "LogLevel" is an optional parameter, i.e. if "LogDirectory" was set, the default value for LogLevel is LogAlways.
+        Environment.SetEnvironmentVariable("LogLevel", "LogAlways");
+    }
+    ```
+
+#### To disable self-diagnostics
+
+Depending on how self-diagnostcis was enabled, delete the configuration file
+or delete the enviornment variable(s).
 
 #### Configuration Parameters
 
 A `FileSize`-KiB log file named as `YearMonthDay-HourMinuteSecond.ExecutableName.ProcessId.log` (e.g. `20010101-120000.foobar.exe.12345.log`) will be generated at the specified directory `LogDirectory`.
 The file name starts with the `DateTime.UtcNow` timestamp of when the file was created.
 
-1. `LogDirectory` is the directory where the output log file will be stored. 
+1. `LogDirectory` is the directory where the output log file will be stored.
 It can be an absolute path or a relative path to the current directory.
 
 2. `FileSize` is a positive integer, which specifies the log file size in [KiB](https://en.wikipedia.org/wiki/Kibibyte).
 This value must be between 1 MiB and 128 MiB (inclusive), or it will be rounded to the closest upper or lower limit.
-The log file will never exceed this configured size, and will be circularly rewriten.
+The log file will never exceed this configured size, and will be circularly rewriten. For the case that configuration
+was passed in via enviornment variables, the default value is `1 MiB` if this parameter was not set.
```

Comment:
Is there the code for setting default value if missing?

Reply:
yup, it's here:

https://github.com/microsoft/ApplicationInsights-dotnet/blob/5b4fd3eccc58a8dd450ea58aa0c538302f31da04/BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigParser.cs#L118